### PR TITLE
[Tags] Add Tag::getByPath() and Tag::__toString()

### PR DESCRIPTION
**Pull request by BlackbitNeueMedien**
Similar to objects, assets and documents this PR adds the methods `Tag::getByPath()` and `Tag::__toString()`

### CODE REVIEW

File: models/Element/Tag.php [hunk 1]
```diff
@@ -160,6 +160,10 @@ public static function getElementsForTag(
         return $tag->getDao()->getElementsForTag($tag, $type, $subtypes, $classNames, $considerChildTags);
     }
 
+    public static function getByPath($path) {
```
Comment (dvesh3) [inline models/Element/Tag.php:163]: please rename it to `getByNamePath` since it works only with name path.

File: models/Element/Tag/Dao.php [hunk 2]
```diff
@@ -292,4 +292,38 @@ public function getElementsForTag(
...
+    /**
+     * @param string $tagPath separated by "/"
+     *
+     * @return null|Tag
+     */
+    public function getByPath($tagPath) {
+        $parentTagId = 0;
+
+        $tag = null;
+        foreach (explode('/', $tagPath) as $tagItem) {
```
Comment (dvesh3) [inline models/Element/Tag/Dao.php:305]: It should work with same value return by `getNamePath()` which is with leading slash e.g. `"/key1/key2/key2"`

File: models/Element/Tag.php (hunk 1 above)
Comment (BlackbitNeueMedien) [reply models/Element/Tag.php:163]: Wanted to use the same name as for data objects, assets etc. There it is also named `getByPath()` and not `getByFilenamePath()` for assets and `getByKeyPath()` for data objects.
Please confirm that I really should use a different name here.

File: models/Element/Tag.php [hunk 3]
```diff
@@ -160,6 +160,10 @@ public static function getElementsForTag(
...
+        return (new self)->getDao()->getByPath($path);
```
[the remark this answers is not in the document]
Comment (BlackbitNeueMedien) [reply models/Element/Tag.php]: Fixed.

File: models/Element/Tag/Dao.php (hunk 2 above)
Comment (BlackbitNeueMedien) [reply models/Element/Tag/Dao.php:306]: Fixed.

File: models/Element/Tag.php (hunk 1 above)
Comment (dvesh3) [reply models/Element/Tag.php:168]: Ok, let's keep the name `getByPath()` for consistency. At least, add doc stating that it works only with name path. thanks!

Comment (BlackbitNeueMedien) [reply models/Element/Tag.php:168]: Fixed.

**Pull request merged**